packages/asset-swapper/src/utils/market_operation_utils/fills.ts

```diff
@@ -71,19 +72,26 @@ function nativeOrdersToPath(
         const input = side === MarketOperation.Sell ? takerAmount : makerAmount;
         const output = side === MarketOperation.Sell ? makerAmount : takerAmount;
         const penalty = ethToOutputRate.times(fees[ERC20BridgeSource.Native] || 0);
-        const adjustedOutput = side === MarketOperation.Sell ? output.minus(penalty) : output.plus(penalty);
         const rate = makerAmount.div(takerAmount);
+        // targetInput can be less than the order size
+        // whilst the penalty is constant, it affects the adjusted output
+        // only up until the target has been exhausted.
+        // A large order and an order at the exact target should be penalized
```

Comment:
should *not* be penalized the same?

Reply:
they **should** be penalized the same, since the cost is constant and the amount is constant. Where having a bigger order size is no advantage.